[CORL-234] Show user's account history in the user drawer

nick-funk opened the pull request:
## What does this PR do?

Adds an account history tab to the user drawer.

## How do I test this PR?

- Head to the moderate section signed in as a moderate level user
- Click on a username on a moderate card or in the community section
- Click on the account history tab in the user drawer
- See history of bans

Notes:

- Suspension records will not exist as we can't create suspensions yet

nick-funk committed:
Create unit tests around the user drawer

CORL-443

nick-funk committed:
Update comment fixture generation to include reason metadata, action counts

CORL-443

nick-funk committed:
Add a user status drop down to the user history drawer

CORL-227

nick-funk committed:
Refactor UserHistoryDrawer tabs and queries for consistency

CORL-234

nick-funk committed:
Create and empty account history tab in the user drawer

CORL-234

nick-funk committed:
Create preliminary account history tab queries

Massages the suspension and ban history together into a usable format.
Sorts the ban and suspension history by date, ascending.
Ready to process the history data into components to make up the future
table records.

CORL-234

nick-funk committed:
Style the user drawer account history records into table format

CORL-234

nick-funk committed:
Fix typo in admin localization around banned text

CORL-234

nick-funk committed:
Update suspension record to support templated insertion of timespan

CORL-234

nick-funk committed:
Create localized parsing of suspension times for account history

CORL-234

nick-funk committed:
Handle errors when loading the user drawer account history

CORL-234

nick-funk committed:
Adjust margins of tab content relative to the tab controls

CORL-234

nick-funk committed:
Extract suspension record time calculations out of component

CORL-234

nick-funk committed:
Handle when a user has no account history

CORL-234

nick-funk committed:
Adjust line height to prevent cutting off usernames in community table

CORL-234

nick-funk committed:
Override hover behaviour for user drawer account history rows

CORL-234

nick-funk committed:
Adjust default column widths in user drawer - account history

CORL-234

wyattjoh committed:
Merge branch 'next' into feature/CORL-234

wyattjoh commented on src/core/client/admin/components/UserHistoryDrawer/BanRecord.tsx:
```diff
@@ -0,0 +1,49 @@
+  active,
+  createdBy,
+}) => {
+  let action: any = null;
```
```suggestion
  let action: React.ReactNode = null;
```

wyattjoh commented on src/core/client/admin/components/UserHistoryDrawer/Tabs.tsx:
```diff
@@ -4,19 +4,20 @@ import React, { FunctionComponent, useCallback, useState } from "react";
   userID: string;
 }
 
 const UserHistoryTabs: FunctionComponent<Props> = ({ userID }) => {
```
Suggest renaming this `UserTabs`.

wyattjoh commented on src/core/client/admin/components/UserHistoryDrawer/UserHistoryDrawerAccountHistory.tsx:
```diff
@@ -0,0 +1,155 @@
+import BanRecord from "./BanRecord";
+import SuspensionRecord from "./SuspensionRecord";
+
+import { Localized } from "fluent-react/compat";
```
Import ordering

wyattjoh commented on src/core/client/admin/components/UserHistoryDrawer/UserHistoryDrawerAccountHistory.tsx:
```diff
@@ -0,0 +1,155 @@
+  createdBy: string | null | undefined;
+  createdAt: Date;
+  active: boolean;
+  from: From | undefined;
```
```suggestion
  from?: From;
```

wyattjoh commented on src/core/client/admin/components/UserHistoryDrawer/UserHistoryDrawerAccountHistoryQuery.tsx:
```diff
@@ -0,0 +1,67 @@
+
+import { CallOut, Spinner } from "coral-ui/components";
+
+import { UserHistoryDrawerAccountHistoryQuery as QueryTypes } from "coral-admin/__generated__/UserHistoryDrawerAccountHistoryQuery.graphql";
```
Import ordering

wyattjoh commented on src/core/client/admin/components/UserHistoryDrawer/UserHistoryDrawerAccountHistoryQuery.tsx:
```diff
@@ -0,0 +1,67 @@
+  userID: string;
+}
+
+const UserHistoryDrawerAccountHistoryQuery: FunctionComponent<Props> = ({
```
Suggest renaming `UserDrawerAccountHistoryQuery`.

wyattjoh commented on src/core/client/admin/components/UserHistoryDrawer/UserHistoryDrawerAllComments.tsx:
```diff
@@ -9,21 +9,21 @@ import { graphql, RelayPaginationProp } from "react-relay";
-import { UserHistoryAllComments_user } from "coral-admin/__generated__/UserHistoryAllComments_user.graphql";
-import { UserHistoryAllComments_viewer } from "coral-admin/__generated__/UserHistoryAllComments_viewer.graphql";
-import { UserHistoryAllCommentsPaginationQueryVariables } from "coral-admin/__generated__/UserHistoryAllCommentsPaginationQuery.graphql";
+import { UserHistoryDrawerAllComments_settings } from "coral-admin/__generated__/UserHistoryDrawerAllComments_settings.graphql";
```
Import ordering

wyattjoh commented on src/core/client/admin/components/UserHistoryDrawer/UserHistoryDrawerQuery.tsx:
```diff
@@ -78,20 +78,10 @@ const UserHistoryDrawerQuery: FunctionComponent<Props> = ({
-              </Flex>
+              <div className={styles.userStatusLabel}>Status:</div>
+              <div className={styles.userStatusChange}>
+                <UserStatusChangeContainer user={user} fullWidth={false} />
```
Prefer a `fullWidth` that defaults to false rather than needing to specify `fullWidth={false}`

wyattjoh commented on src/core/client/admin/components/UserHistoryDrawer/UserHistoryDrawerRejectedComments.tsx:
```diff
@@ -9,21 +9,21 @@ import { graphql, RelayPaginationProp } from "react-relay";
+import { UserHistoryDrawerRejectedComments_settings } from "coral-admin/__generated__/UserHistoryDrawerRejectedComments_settings.graphql";
+import { UserHistoryDrawerRejectedComments_user } from "coral-admin/__generated__/UserHistoryDrawerRejectedComments_user.graphql";
+import { UserHistoryDrawerRejectedComments_viewer } from "coral-admin/__generated__/UserHistoryDrawerRejectedComments_viewer.graphql";
+import { UserHistoryDrawerRejectedCommentsPaginationQueryVariables } from "coral-admin/__generated__/UserHistoryDrawerRejectedCommentsPaginationQuery.graphql";
```
Import ordering

wyattjoh commented on src/core/client/admin/components/UserHistoryDrawer/UserHistoryDrawerRejectedCommentsQuery.tsx:
```diff
@@ -4,32 +4,32 @@ import { ReadyState } from "react-relay";
 import { CallOut, Spinner } from "coral-ui/components";
 
-import { UserHistoryRejectedCommentsContainerQuery as QueryTypes } from "coral-admin/__generated__/UserHistoryRejectedCommentsContainerQuery.graphql";
+import { UserHistoryDrawerRejectedCommentsQuery as QueryTypes } from "coral-admin/__generated__/UserHistoryDrawerRejectedCommentsQuery.graphql";
```
Import ordering

wyattjoh commented on src/core/client/admin/components/UserHistoryDrawer/SuspensionRecord.tsx:
```diff
@@ -0,0 +1,185 @@
+        <Localized id={"moderate-user-drawer-account-history-suspension-time"}>
+          Suspension
+        </Localized>
+        {timeSpan}
```
This provides an issue from a translation perspective. It should not be split into two parts, instead preferring a single `Suspension, 2 weeks`.

nick-funk committed:
Use react node type instead of any for component substitution

CORL-234

nick-funk committed:
Rename TabType to UserTabs

CORL-234

nick-funk committed:
Use nullable type instead of strict typing using OR's

CORL-234

nick-funk committed:
Set UserStatusChangeContainer fullWidth default to false

CORL-234

nick-funk committed:
Combine timed suspension localization into one Localized element

CORL-234

wyattjoh committed:
fix: useMemo on history bar

nick-funk committed:
Rename UserHistoryDrawerAccountHistory to UserDrawerAccountHistory

Cleans up the agglutination's on these file names.

Probably means we should rename UserHistoryDrawer... to
UserDrawer... everywhere. Will get to that later.

CORL-234

nick-funk committed:
Reorder imports to match alphabetization preferences

CORL-234

wyattjoh committed:
feat: simplication to action table

wyattjoh committed:
feat: visual tweaks

wyattjoh committed:
chore: added notes for deprecating

wyattjoh committed:
Merge branch 'next' into feature/CORL-234

wyattjoh committed:
Merge branch 'next' into feature/CORL-234
